timesketch/api/v1/resources.py

```diff
@@ -2148,6 +2158,84 @@ def get(self, searchindex_id):
         return self.to_json(searchindex)
 
 
+class UserListResource(ResourceMixin, Resource):
+    """Resource to get list of users."""
+
+    @login_required
+    def get(self):
+        """Handles GET request to the resource.
+
+        Returns:
+            List of usernames
+        """
+        return self.to_json(User.query.all())
+
+
+class GroupListResource(ResourceMixin, Resource):
+    """Resource to get list of groups."""
+
+    @login_required
+    def get(self):
+        """Handles GET request to the resource.
+
+        Returns:
+            List of group names
+        """
+        return self.to_json(Group.query.all())
+
+
+class CollaboratorResource(ResourceMixin, Resource):
+    """Resource to update sketch collaborators."""
+
+    @login_required
+    def post(self, sketch_id):
+        """Handles POST request to the resource.
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        form = request.json
+
+        if not sketch.has_permission(user=current_user, permission='write'):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN,
+                'The user does not have write permission on the sketch.')
+
+        for username in form.get('users', []):
+            base_username = username.split('@')[0]
+            base_username = base_username.strip()
+            user = User.query.filter_by(username=base_username).first()
+            if user:
+                sketch.grant_permission(permission='read', user=user)
+                sketch.grant_permission(permission='write', user=user)
+
+        for group in form.get('groups', []):
+            group = Group.query.filter_by(name=group).first()
+            # Only add groups publicly visible or owned by the current user
+            if not group.user or group.user == current_user:
+                sketch.grant_permission(permission='read', group=group)
+                sketch.grant_permission(permission='write', group=group)
+
+        for username in form.get('remove_users', []):
+            user = User.query.filter_by(username=username).first()
+            sketch.revoke_permission(permission='read', user=user)
```

Comment:
if more granular, you would have to then check permissions first and remove those that exist

Reply:
yes. Opened up an issue for it: https://github.com/google/timesketch/issues/1016